### ISSUE TITLE
ViewState generation with View interface parametrized by "T extends SomeInterface"

### ISSUE THREAD

**Issue by Seerus** [Arello-Mobile/Moxy#98]
Hi all!
I have 3 simple classes:
1. Class for base list item view:
```
public interface ListItemView<T> extends BaseView {
  void setModel(final T model);
}
```
2. Child of previous class, but is parametrized by interface: 
```
public interface MessageListView<Model extends Message> extends ListItemView<Model> {
...
}
```
3. Interface for previous class: 
```
public interface Message extends Comparable<Message>  {
 ...
}
```

When i try to rebuild my project, `moxy-compiler` generated code with this method in `ListItemView$$ViewState` class:
```
public  void setModel(null model) {
...
}
```

I have `null`-class argument except of `Model`

**Comment by Seerus** [comment]
I think a problem in `ViewStateClassGenerator` in `moxy-compiler` module. 
In cases like my `ExecutableType.asType().getTypeVariables()` return empty list, and so `methodTypes` Map has no keys for all arguments (lines 259-269 of `ViewStateClassGenerator`). 
In line 295 `parameters` have 1 item with type `Model`, as i except. 

In `Util` in line 84 this param is instance of `TypeVariable`, but `types.get(param.toString())` return null for the reasons described above. So type is filled by `null` class

**Comment by Seerus** [comment]
I found one solution. In theory, nothing should break down

**Comment by senneco** [comment]
This is published in version 1.4.6. Thanks to pull request ;)
